libfwupdplugin/fu-self-test.c

```diff
@@ -1329,6 +1330,7 @@ fu_plugin_vfuncs_func(void)
 static void
 fu_plugin_device_gtype_func(void)
 {
+#ifdef HAVE_LIBUSB
```

Comment:
can you avoid all the ifdef related to this test by just one small ifdef to skip the test if not HAVE_LIBUSB?

Reply:
Nah, we don't compile fu-usb-device.c in that case (to avoid *hundreds* of other ifdefs!) so FU_TYPE_USB_DEVICE isn't defined. But I guess I can use FU_TYPE_UDEV_DEVICE instead -- so fixed :)

Reply:
Actually, the right thing to do is https://github.com/fwupd/fwupd/pull/7439 -- which means it's testing the right thing and won't be affected by udev-less targets.